learn/building_programs.md

```diff
@@ -0,0 +1,780 @@
+---
+layout: page
+title: Building programs
+permalink: /learn/build_programs
+navbar: Learn
+---
+
+
+
+Draft tutorial: Building programs
+---------------------------------
+
+Remark: I only included Linux and Windows as operating systems in this
+tutorial, for the simple reason that I have no experience with MacOS*
+in any of its guises. While MacOS* is very similar to Linux, it is the
+differences that may need to be documented. Some information would be
+helpful.

```

Comment:
```suggestion
Remark: this tutorial gives examples for the Windows and Linux operating systems,
however the workflow and general principles still apply to MacOS.
```

It would be good if someone with experience developing on MacOS is able to check the tutorial content for any significant differences, especially with libraries.
If not, then maybe we can link to a good online resource for this?